src/prefect/orion/schemas/states.py

```diff
@@ -52,11 +54,37 @@ class StateDetails(PrefectBaseModel):
     refresh_cache: bool = None
 
 
-class State(IDBaseModel, Generic[R]):
+class StateBaseModel(IDBaseModel):
+    def orm_dict(
+        self, *args, shallow: bool = False, json_compatible: bool = False, **kwargs
+    ) -> dict:
+        schema_dict = self.dict(
+            *args, shallow=shallow, json_compatible=json_compatible, **kwargs
+        )
+        # remove the data field in order to construct a state ORM model
+        schema_dict.pop("data", None)
+        return schema_dict
+
+
+class StateGetter(GetterDict):
+    def get(self, key: str, default: Any) -> Any:
+        # During orchestration, ORM states can be instantiated prior to inserting
+        # results into the artifact table, and the `data` field will fail to load
+        if key == "data":
+            try:
+                return super().get(key, default)
+            except MissingGreenlet:
+                return None
+
+        return super().get(key, default)
+
+
+class State(StateBaseModel, Generic[R]):
```

Comment:
We should be able to drop the generic — that's only necessary client-side now.

Reply:
It seems to break everything when I remove the generic so I'm leaving it for now